--- vtun-3.0.1.orig6/Makefile.in	2007-10-13 19:47:28.000000000 +0200
+++ vtun-3.0.1/Makefile.in	2007-10-13 19:51:56.000000000 +0200
@@ -90,10 +90,18 @@
 	fi
 
 install: vtund install_config install_man
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(VAR_DIR)/run
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(STAT_DIR)
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(LOCK_DIR)
-	$(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(SBIN_DIR)
+	if [ ! -r $(DESTDIR)$(VAR_DIR)/run ]; then \
+	  $(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(VAR_DIR)/run; \
+	fi
+	if [ ! -r $(DESTDIR)$(STAT_DIR) ]; then \
+	  $(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(STAT_DIR); \
+	fi
+	if [ ! -r $(DESTDIR)$(LOCK_DIR) ]; then \
+	  $(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(LOCK_DIR); \
+	fi
+	if [ ! -r $(DESTDIR)$(SBIN_DIR) ]; then \
+	  $(INSTALL) -d -m 755 $(INSTALL_OWNER) $(DESTDIR)$(SBIN_DIR); \
+	fi
 	$(INSTALL) -m 755 $(INSTALL_OWNER) vtund $(DESTDIR)$(SBIN_DIR)
 
 # DO NOT DELETE THIS LINE -- make depend depends on it.
