--- linux-2.6.23.1.orig/kernel/time/Makefile	2007-10-09 22:31:38.000000000 +0200
+++ linux-2.6.23.1/kernel/time/Makefile	2007-10-13 23:40:58.000000000 +0200
@@ -6,3 +6,7 @@
 obj-$(CONFIG_TICK_ONESHOT)			+= tick-oneshot.o
 obj-$(CONFIG_TICK_ONESHOT)			+= tick-sched.o
 obj-$(CONFIG_TIMER_STATS)			+= timer_stats.o
+
+# workaround for gcc.gnu.org/PR32044
+CFLAGS_timekeeping.o += $(call cc-ifversion, -ge, 0403, -fno-tree-scev-cprop)
+
