Index: gcc-4.3.0/libgcc/config/libbid/bid_conf.h
===================================================================
--- gcc-4.3.0/libgcc/config/libbid/bid_conf.h	(revision 129202)
+++ gcc-4.3.0/libgcc/config/libbid/bid_conf.h	(working copy)
@@ -535,7 +535,9 @@ Software Foundation, 51 Franklin Street,
 #define DECIMAL_GLOBAL_ROUNDING 1
 #define DECIMAL_GLOBAL_ROUNDING_ACCESS_FUNCTIONS 1
 #define DECIMAL_GLOBAL_EXCEPTION_FLAGS 1
+#ifndef __GCC_FLOAT_NOT_NEEDED
 #define DECIMAL_GLOBAL_EXCEPTION_FLAGS_ACCESS_FUNCTIONS 1
+#endif
 #define BID_HAS_GCC_DECIMAL_INTRINSICS 1
 #endif /* IN_LIBGCC2 */
 
