--- openssl-0.9.8h.orig/apps/speed.c	2008-06-08 11:26:57.000000000 +0200
+++ openssl-0.9.8h/apps/speed.c	2008-06-08 11:31:39.000000000 +0200
@@ -117,6 +117,16 @@
 # define TIMEB
 #endif
 
+#if defined __UCLIBC__
+# undef TIMEB
+# undef USE_TOD
+# if defined __UCLIBC_SUSV3_LEGACY__
+#  define TIMEB
+# else
+#  define USE_TOD
+# endif
+#endif
+
 #if defined(OPENSSL_SYS_NETWARE)
 #undef TIMES
 #undef TIMEB
