Index: uClibc/utils/Makefile.in
===================================================================
--- uClibc/utils/Makefile.in	(revision 21931)
+++ uClibc/utils/Makefile.in	(working copy)
@@ -96,7 +96,20 @@ ifeq ($(UCLIBC_HAS_LOCALE),y)
 	$(INSTALL) -m 755 $(utils_OUT)/locale $(PREFIX)$(RUNTIME_PREFIX)usr/bin/locale
 endif
 
+hostutils_install: hostutils
+	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)usr/bin
+	#$(INSTALL) -m 755 $(utils_OUT)/readelf.host $(PREFIX)$(RUNTIME_PREFIX)usr/bin/readelf
+ifeq ($(HAVE_SHARED),y)
+	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)sbin
+	$(INSTALL) -m 755 $(utils_OUT)/ldd.host $(PREFIX)$(RUNTIME_PREFIX)usr/bin/ldd
+	$(INSTALL) -m 755 $(utils_OUT)/ldconfig.host $(PREFIX)$(RUNTIME_PREFIX)sbin/ldconfig
+endif
+ifeq ($(UCLIBC_HAS_LOCALE),y)
+	$(INSTALL) -m 755 $(utils_OUT)/iconv.host $(PREFIX)$(RUNTIME_PREFIX)usr/bin/iconv
+	$(INSTALL) -m 755 $(utils_OUT)/locale.host $(PREFIX)$(RUNTIME_PREFIX)usr/bin/locale
+endif
+
 objclean-y += utils_clean
 
 utils_clean:
-	$(RM) $(utils_OUT)/{ldconfig,ldd,readelf,iconv,locale,*.host}
+	$(RM) $(addprefix $(utils_OUT)/,ldconfig ldd readelf iconv locale *.host)
