Index: uClibc/libc/sysdeps/linux/common/Makefile.in
===================================================================
--- uClibc/libc/sysdeps/linux/common/Makefile.in	(revision 23383)
+++ uClibc/libc/sysdeps/linux/common/Makefile.in	(working copy)
@@ -35,8 +35,8 @@ CSRC := $(filter-out inotify.c ioperm.c 
 	modify_ldt.c personality.c prctl.c readahead.c reboot.c \
 	remap_file_pages.c sched_getaffinity.c sched_setaffinity.c \
 	sendfile64.c sendfile.c setfsgid.c setfsuid.c setresuid.c \
-	splice.c vmsplice.c tee.c swapoff.c swapon.c sysctl.c sysinfo.c \
-	uselib.c vhangup.c,$(CSRC))
+	splice.c vmsplice.c tee.c swapoff.c swapon.c sync_file_range.c \
+	sysctl.c sysinfo.c uselib.c vhangup.c,$(CSRC))
 endif
 
 ifneq ($(UCLIBC_BSD_SPECIFIC),y)
Index: uClibc/libc/sysdeps/linux/common/sync_file_range.c
===================================================================
--- uClibc/libc/sysdeps/linux/common/sync_file_range.c	(revision 0)
+++ uClibc/libc/sysdeps/linux/common/sync_file_range.c	(revision 0)
@@ -0,0 +1,33 @@
+/* vi: set sw=4 ts=4: */
+/*
+ * sync_file_range() for uClibc
+ *
+ * Copyright (C) 2008 Bernhard Reutner-Fischer <uclibc@uclibc.org>
+ *
+ * Licensed under the LGPL v2.1, see the file COPYING.LIB in this tarball.
+ */
+
+#include <sys/syscall.h>
+#if defined __USE_GNU
+#include <fcntl.h>
+
+#if defined __NR_sync_file_range && defined __UCLIBC_HAS_LFS__
+#define __NR___syscall_sync_file_range __NR_sync_file_range
+static __inline__ _syscall6(int, __syscall_sync_file_range, int, fd,
+		off_t, offset_hi, off_t, offset_lo,
+		off_t, nbytes_hi, off_t, nbytes_lo, unsigned int, flags);
+int sync_file_range(int fd, off64_t offset, off64_t nbytes, unsigned int flags)
+{
+	return __syscall_sync_file_range(fd,
+		__LONG_LONG_PAIR((long)(offset >> 32), (long)(offset & 0xffffffff)),
+		__LONG_LONG_PAIR((long)(nbytes >> 32), (long)(nbytes & 0xffffffff)),
+		flags);
+}
+#elif defined __UCLIBC_HAS_STUBS__
+int sync_file_range(int fd, off64_t offset, off64_t nbytes, unsigned int flags)
+{
+	__set_errno(ENOSYS);
+	return -1;
+}
+#endif
+#endif
Index: uClibc/libc/sysdeps/linux/alpha/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/alpha/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/alpha/bits/fcntl.h	(working copy)
@@ -182,7 +182,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -193,7 +192,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -213,11 +211,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/arm/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/arm/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/arm/bits/fcntl.h	(working copy)
@@ -190,7 +190,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -201,7 +200,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -220,11 +218,9 @@ __BEGIN_DECLS
 extern ssize_t readahead (int __fd, __off64_t __offset, size_t __count)
     __THROW;
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/bfin/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/bfin/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/bfin/bits/fcntl.h	(working copy)
@@ -186,7 +186,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -197,7 +196,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -217,11 +215,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/cris/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/cris/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/cris/bits/fcntl.h	(working copy)
@@ -187,7 +187,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -198,7 +197,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -218,11 +216,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/e1/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/e1/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/e1/bits/fcntl.h	(working copy)
@@ -183,7 +183,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -194,7 +193,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -214,11 +212,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/frv/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/frv/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/frv/bits/fcntl.h	(working copy)
@@ -165,7 +165,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -176,7 +175,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -196,11 +194,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/h8300/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/h8300/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/h8300/bits/fcntl.h	(working copy)
@@ -183,7 +183,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -194,7 +193,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -214,11 +212,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/hppa/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/hppa/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/hppa/bits/fcntl.h	(working copy)
@@ -177,7 +177,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -188,7 +187,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -208,11 +206,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/i386/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/i386/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/i386/bits/fcntl.h	(working copy)
@@ -190,7 +190,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -201,7 +200,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -221,11 +219,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/i960/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/i960/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/i960/bits/fcntl.h	(working copy)
@@ -183,7 +183,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -194,7 +193,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -214,11 +212,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/ia64/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/ia64/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/ia64/bits/fcntl.h	(working copy)
@@ -184,7 +184,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -195,7 +194,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -215,11 +213,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/m68k/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/m68k/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/m68k/bits/fcntl.h	(working copy)
@@ -186,7 +186,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -197,7 +196,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -217,11 +215,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/microblaze/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/microblaze/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/microblaze/bits/fcntl.h	(working copy)
@@ -183,7 +183,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -194,7 +193,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -214,11 +212,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/mips/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/mips/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/mips/bits/fcntl.h	(working copy)
@@ -210,7 +210,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -221,7 +220,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -241,11 +239,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/nios2/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/nios2/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/nios2/bits/fcntl.h	(working copy)
@@ -186,7 +186,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -197,7 +196,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -217,11 +215,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/nios/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/nios/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/nios/bits/fcntl.h	(working copy)
@@ -186,7 +186,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -197,7 +196,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -217,11 +215,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/powerpc/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/powerpc/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/powerpc/bits/fcntl.h	(working copy)
@@ -190,7 +190,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -201,7 +200,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -221,11 +219,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/sh64/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/sh64/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/sh64/bits/fcntl.h	(working copy)
@@ -165,7 +165,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -176,7 +175,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -196,11 +194,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/sh/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/sh/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/sh/bits/fcntl.h	(working copy)
@@ -190,7 +190,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -201,7 +200,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -221,11 +219,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/sparc/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/sparc/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/sparc/bits/fcntl.h	(working copy)
@@ -209,7 +209,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -220,7 +219,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -240,11 +238,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/v850/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/v850/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/v850/bits/fcntl.h	(working copy)
@@ -183,7 +183,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -194,7 +193,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -214,11 +212,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/vax/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/vax/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/vax/bits/fcntl.h	(working copy)
@@ -163,7 +163,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -174,7 +173,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -194,11 +192,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/x86_64/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/x86_64/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/x86_64/bits/fcntl.h	(working copy)
@@ -204,7 +204,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -215,7 +214,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -235,11 +233,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
Index: uClibc/libc/sysdeps/linux/xtensa/bits/fcntl.h
===================================================================
--- uClibc/libc/sysdeps/linux/xtensa/bits/fcntl.h	(revision 23383)
+++ uClibc/libc/sysdeps/linux/xtensa/bits/fcntl.h	(working copy)
@@ -187,7 +187,6 @@ struct flock64
 
 
 #ifdef __USE_GNU
-#if 0
 /* Flags for SYNC_FILE_RANGE.  */
 # define SYNC_FILE_RANGE_WAIT_BEFORE	1 /* Wait upon writeout of all pages
 					     in the range before performing the
@@ -198,7 +197,6 @@ struct flock64
 # define SYNC_FILE_RANGE_WAIT_AFTER	4 /* Wait upon writeout of all pages in
 					     the range after performing the
 					     write.  */
-#endif
 
 /* Flags for SPLICE and VMSPLICE.  */
 # define SPLICE_F_MOVE		1	/* Move pages instead of copying.  */
@@ -218,11 +216,9 @@ extern ssize_t readahead (int __fd, __of
     __THROW;
 
 
-#if 0
 /* Selective file content synch'ing.  */
 extern int sync_file_range (int __fd, __off64_t __from, __off64_t __to,
 			    unsigned int __flags);
-#endif
 
 /* Splice address range into a pipe.  */
 extern ssize_t vmsplice (int __fdout, const struct iovec *__iov,
