Index: uClibc/Rules.mak
===================================================================
--- uClibc/Rules.mak	(revision 20213)
+++ uClibc/Rules.mak	(working copy)
@@ -340,6 +340,7 @@
 
 # Add a bunch of extra pedantic annoyingly strict checks
 XWARNINGS=$(subst ",, $(strip $(WARNINGS))) -Wstrict-prototypes -fno-strict-aliasing
+# ")
 ifeq ($(EXTRA_WARNINGS),y)
 XWARNINGS+=-Wnested-externs -Wshadow -Wmissing-noreturn -Wmissing-format-attribute -Wformat=2
 XWARNINGS+=-Wmissing-prototypes -Wmissing-declarations
@@ -348,7 +349,9 @@
 #XWARNINGS+=-Wdeclaration-after-statement
 endif
 XARCH_CFLAGS=$(subst ",, $(strip $(ARCH_CFLAGS)))
+# ")
 CPU_CFLAGS=$(subst ",, $(strip $(CPU_CFLAGS-y)))
+# ")
 
 SSP_DISABLE_FLAGS ?= $(call check_gcc,-fno-stack-protector,)
 ifeq ($(UCLIBC_BUILD_SSP),y)
@@ -366,8 +369,14 @@
 
 ifneq ($(strip $(UCLIBC_EXTRA_CFLAGS)),"")
 CFLAGS += $(subst ",, $(UCLIBC_EXTRA_CFLAGS))
+# ")
 endif
 
+ifneq ($(strip $(UCLIBC_EXTRA_LDFLAGS)),"")
+LDFLAGS += $(subst ",, $(UCLIBC_EXTRA_LDFLAGS))
+# ")
+endif
+
 LDADD_LIBFLOAT=
 ifeq ($(UCLIBC_HAS_SOFT_FLOAT),y)
 # If -msoft-float isn't supported, we want an error anyway.
