diff -rdup pciutils-2.2.7.orig/lib/configure pciutils-2.2.7/lib/configure
--- pciutils-2.2.7.orig/lib/configure	2007-09-12 21:24:40.000000000 +0200
+++ pciutils-2.2.7/lib/configure	2007-10-15 20:24:53.000000000 +0200
@@ -12,7 +12,7 @@ echo_n() {
 echo_n "Configuring libpci for your system..."
 idsdir=${1:-/usr/share}
 version=${2:-0.0}
-sys=`uname -s`
+sys=Linux
 rel=`uname -r`
 if [ "$sys" = "AIX" -a -x /usr/bin/oslevel -a -x /usr/sbin/lsattr ]
 then
Only in pciutils-2.2.7/lib: configure.orig
