diff -rdup ltp-full-20080831.oorig/testcases/kernel/mem/hugetlb/lib/Makefile ltp-full-20080831/testcases/kernel/mem/hugetlb/lib/Makefile
--- ltp-full-20080831.oorig/testcases/kernel/mem/hugetlb/lib/Makefile	2008-08-31 16:33:03.000000000 +0200
+++ ltp-full-20080831/testcases/kernel/mem/hugetlb/lib/Makefile	2008-09-26 17:17:17.000000000 +0200
@@ -21,7 +21,12 @@
 # description	: make(1) description file to build a library for the 	#
 #		  common routines in the ipc(2) tests.			#
 #########################################################################
+
+ifeq ($(UCLIBC_SUSV3_LEGACY),y)
 SRCS=libipc.c
+else
+SRCS=
+endif
 OBJS=$(SRCS:.c=.o)
 MAINS=libipc.a
 CFLAGS+=-I../../../../../include -g -Wall
Only in ltp-full-20080831/testcases/kernel/mem/hugetlb/lib: Makefile~
diff -rdup ltp-full-20080831.oorig/testcases/kernel/sched/tool/Makefile ltp-full-20080831/testcases/kernel/sched/tool/Makefile
--- ltp-full-20080831.oorig/testcases/kernel/sched/tool/Makefile	2008-08-31 16:33:19.000000000 +0200
+++ ltp-full-20080831/testcases/kernel/sched/tool/Makefile	2008-09-26 17:16:57.000000000 +0200
@@ -2,6 +2,10 @@ CFLAGS += -Wall -O -g
 LDLIBS := -lpthread
 
 SRCS=$(wildcard *.c)
+ifneq ($(UCLIBC_SUSV3_LEGACY),y)
+SRCS:=$(filter-out trace_sched.c,$(SRCS))
+endif
+
 TARGETS=$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
Only in ltp-full-20080831/testcases/kernel/sched/tool: Makefile~
diff -rdup ltp-full-20080831.oorig/testcases/kernel/syscalls/ipc/lib/Makefile ltp-full-20080831/testcases/kernel/syscalls/ipc/lib/Makefile
--- ltp-full-20080831.oorig/testcases/kernel/syscalls/ipc/lib/Makefile	2008-08-31 16:33:16.000000000 +0200
+++ ltp-full-20080831/testcases/kernel/syscalls/ipc/lib/Makefile	2008-09-26 17:16:31.000000000 +0200
@@ -16,7 +16,11 @@
 #  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #
 
+ifeq ($(UCLIBC_SUSV3_LEGACY),y)
 SRCS   = libipc.c
+else
+SRCS   =
+endif
 OBJS   = $(SRCS:.c=.o)
 LIBIPC = ../libipc.a
 LIBIPC_HEADERS	= ipcmsg.h ipcsem.h
Only in ltp-full-20080831/testcases/kernel/syscalls/ipc/lib: Makefile~
