diff --exclude=Makefile -rdup argus-2.0.6.fixes.1.orig/common/gencode.c argus-2.0.6.fixes.1/common/gencode.c
--- argus-2.0.6.fixes.1.orig/common/gencode.c	2004-02-23 16:00:36.000000000 +0100
+++ argus-2.0.6.fixes.1/common/gencode.c	2008-03-31 15:54:09.000000000 +0200
@@ -58,9 +58,9 @@
 #include <stdarg.h>
 #include <stdlib.h>
 #include <syslog.h>
-
+/*
 #include <net/bpf.h>
-
+*/
 #include <argus_out.h>
 #include <argus_filter.h>
 #include <argus_def.h>
diff --exclude=Makefile -rdup argus-2.0.6.fixes.1.orig/include/argus_filter.h argus-2.0.6.fixes.1/include/argus_filter.h
--- argus-2.0.6.fixes.1.orig/include/argus_filter.h	2004-02-23 16:00:36.000000000 +0100
+++ argus-2.0.6.fixes.1/include/argus_filter.h	2008-03-31 15:54:33.000000000 +0200
@@ -67,7 +67,7 @@
 #include <string.h>
 #include <stdarg.h>
 #include <stdio.h>
-
+#include <pcap-bpf.h>
 #include <gencode.h>
 
 #ifndef __GNUC__
